### PR TITLE
Message/Transaction dialog

### PR DESCRIPTION
**Pull request by mikkio-j**
Issue: #372 
Figma: [https://www.figma.com/file/rBjfm2tJ2Pr9M0UNTduU9v/Joystream-Atlas?node-id=3914%3A1856](https://www.figma.com/file/rBjfm2tJ2Pr9M0UNTduU9v/Joystream-Atlas?node-id=3914%3A1856)

### CODE REVIEW

File: src/components/Dialogs/MessageDialog/MessageDialog.style.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  color: ${colors.gray[300]};
+  word-wrap: break-word;
+`
+
+export const Spinner = styled.div`
+  position: relative;
+  display: inline-block;
+  border-radius: 50%;
+  width: 32px;
+  height: 32px;
```
Comment (drillprop) [inline src/components/Dialogs/MessageDialog/MessageDialog.style.ts]:
```suggestion
  width: ${sizes(8)};
  height: ${sizes(8)};
```

File: src/components/Dialogs/MessageDialog/MessageDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+import ActionDialog, { ActionDialogProps } from '../ActionDialog/ActionDialog'
+import { StyledIcon, StyledTitleText, StyledDescriptionText, Spinner } from './MessageDialog.style'
+import * as Icons from '@/shared/icons'
```
Comment (drillprop) [inline src/components/Dialogs/MessageDialog/MessageDialog.tsx]:
```suggestion
import { IconType } from '@/shared/components/Icon'
```

File: src/components/Dialogs/MessageDialog/MessageDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+} & ActionDialogProps
+
+const MessageDialog: React.FC<MessageDialogProps> = ({
+  title,
+  description,
+  variant = 'info',
+  ...actionDialogProps
+}) => {
+  const hasIcon = ['success', 'warning', 'error'].includes(variant)
+  const icon = `dialog-${variant}` as Icons.IconType
```
Comment (drillprop) [inline src/components/Dialogs/MessageDialog/MessageDialog.tsx]:
```suggestion
  const icon = `dialog-${variant}` as IconType
```

File: src/shared/icons/dialog-error.svg [hunk 4]
```diff
@@ -0,0 +1 @@
+<svg width="32px" height="32px" fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><circle cx="16" cy="16" r="15" stroke="#FF3861" stroke-width="2"/><path d="M11 21l10-10M21 21L11 11" stroke="#FF3861" stroke-width="2"/></svg>
```
Comment (drillprop) [inline src/shared/icons/dialog-error.svg]:
```suggestion
<svg width="32px" height="32px" fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><circle cx="16" cy="16" r="15" stroke="#FF3861" stroke-width="2"/><path d="M11 21l10-10M21 21L11 11" stroke="#FF3861" stroke-width="2"/></svg>

```

File: src/shared/icons/dialog-success.svg [hunk 5]
```diff
@@ -0,0 +1 @@
+<svg  width="32px" height="32px"  fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><circle cx="16" cy="16" r="15" stroke="#00DBB0" stroke-width="2"/><path d="M11 17l3 3 7-8" stroke="#00DBB0" stroke-width="2"/></svg>
```
Comment (drillprop) [inline src/shared/icons/dialog-success.svg]:
```suggestion
<svg  width="32px" height="32px"  fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><circle cx="16" cy="16" r="15" stroke="#00DBB0" stroke-width="2"/><path d="M11 17l3 3 7-8" stroke="#00DBB0" stroke-width="2"/></svg>

```

Comment (drillprop) [reply src/shared/icons/dialog-success.svg]: to fix newline warning 
Zrzut ekranu 2021-03-17 o 10 59 21

**Review by drillprop — COMMENTED** [review]
Looking good.  👍 
In the design, there is also a variant with an image. Shouldn't you add this also?
Zrzut ekranu 2021-03-17 o 11 00 38

File: src/shared/icons/dialog-error.svg (hunk 4 above)
Comment (drillprop) [inline src/shared/icons/dialog-error.svg]: Still not fixed

File: src/shared/icons/dialog-success.svg (hunk 5 above)
Comment (drillprop) [inline src/shared/icons/dialog-success.svg]: Same as above

**Review by drillprop — APPROVED** [review]
Only two things left.

File: src/components/Dialogs/ActionDialog/ActionDialog.style.ts [hunk 6]
```diff
@@ -1,15 +1,25 @@
...
+
 export const ButtonsContainer = styled.div`
   display: flex;
   flex-direction: column;
 
   > * + * {
     margin-top: ${sizes(2)};
   }
+  button {
+    font-weight: ${typography.weights.bold};
```
Comment (kdembler) [inline src/components/Dialogs/ActionDialog/ActionDialog.style.ts]: Looking at Figma all buttons should have 700 weight, so if that's not what's in the component currently, we should update it

File: src/components/Dialogs/ActionDialog/ActionDialog.tsx [hunk 7]
```diff
@@ -29,7 +38,11 @@ const ActionDialog: React.FC<ActionDialogProps> = ({
...
           {additionalActionsNode && <AdditionalActionsContainer>{additionalActionsNode}</AdditionalActionsContainer>}
           <ButtonsContainer>
-            {primaryButtonText && <Button onClick={onPrimaryButtonClick}>{primaryButtonText}</Button>}
+            {primaryButtonText && (
+              <StyledPrimaryButton onClick={onPrimaryButtonClick} warning={warning} error={error}>
+                {primaryButtonText}
+              </StyledPrimaryButton>
+            )}
             {secondaryButtonText && (
               <Button variant="secondary" onClick={onSecondaryButtonClick}>
```
Comment (kdembler) [inline src/components/Dialogs/ActionDialog/ActionDialog.tsx:47]: This button should be transparent, I think you may need to rebase?

File: src/components/Dialogs/TransactionDialog/TransactionDialog.tsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import ActionDialog, { ActionDialogProps } from '../ActionDialog/ActionDialog'
+import { TextContainer, StyledTransactionIllustration, Spinner } from './TransactionDialog.style'
+import { StyledTitleText, StyledDescriptionText } from '../MessageDialog/MessageDialog.style'
+
+export type TransactionDialogProps = {
+  title: string
+  description: string
```
Comment (kdembler) [inline src/components/Dialogs/TransactionDialog/TransactionDialog.tsx]: I think we will always use this dialog in the same context, let's define those right here so we don't need to copy-paste the same thing

Comment (kdembler) [reply src/components/Dialogs/TransactionDialog/TransactionDialog.tsx]: @mikkio-j This wasn't resolved - those values shouldn't be passed as props, but defined here in the component

**Review by kdembler — APPROVED** [review]

**Pull request merged**